%YAML:1.0

#common parameters 
#
# TODO: change these topics 

    #camera calibration 
model_type: PINHOLE
camera_name: camera
image_width: 752 # 640
image_height: 480 # 480
distortion_parameters:
    k1: 0.
    k2: 0.
    p1: 0.
    p2: 0.
projection_parameters: 
        #  615.426, 625.456, 318.837, 240.594
    fx: 457.975
    fy: 457.975
    cx: 467.0
    cy: 235.5

#### feature tracking #### 
sub_region_x: 12 #12
sub_region_y: 8 
max_n_feature_subregion: 4 #2
boundary_x: 20
boundary_y: 20
min_dist_between_features: 20 #20
harris_threshold: 0.0000001 #1e-7
show_feature_tracking_result: 0 #1


#### depth handler ####
cloud_ds_rate: 2
cloud_skip_rate: 1
cloud_max_depth: 5.

#### vio ####
imu_topic: "/imu0"
image_topic: "/rect/first/image" #"/cam0/color" # "/cam0/image_raw"
output_path: "dvio_result.log"          # vins outputs will be wrttento vins_folder_path + output_path

estimate_extrinsic: 0 
ex_calib_result_path: "ex_calib_result.yaml"  # If you choose 1 or 2, the extrinsic calibration result will be written vins_folder_path + ex_calib_result_path.                        
        #If you choose 0 or 1, you should write down the following matrix.
        #Rotation from camera frame to imu frame, imu^R_cam
extrinsicRotation: !!opencv-matrix
    rows: 3
    cols: 3
    dt: d
    data: [0.0148655429818, -0.999880929698, 0.00414029679422, 
        0.999557249008, 0.0149672133247, 0.025715529948, 
        -0.0257744366974, 0.00375618835797, 0.999660727178]
        #Translation from camera frame to imu frame, imu^T_cam
extrinsicTranslation: !!opencv-matrix
    rows: 3
    cols: 1
    dt: d
    # data: [-0.02,-0.06, 0.01]
    # data: [0.063, -0.001, 0.018]
    data: [-0.0216401454975, -0.0646769886768, 0.00981073058949]

        #feature traker paprameters
max_cnt: 150  # 150          # max feature number in feature tracking
min_dist: 30  # 30          # min distance between two features 
freq: 10 # 10                # frequence (Hz) of publish tracking result. At least 10Hz for good estimation. If set 0, the frequence will be same as raw image 
F_threshold: 2.0 # 2.0 # 3.0 # 2.0 1.0        # ransac threshold (pixel)
show_track: 1           # publish tracking image as topic
equalize: 0             # 1 if image is too dark or light, trun on equalize to find enough features
fisheye: 0              # if using fisheye, trun on it. A circle mask will be loaded to remove edge noisy points

        #optimization parameters
max_solver_time: 0.04  # 0.04 max solver itration time (ms), to guarantee real time
max_num_iterations: 8   # 8 max solver itrations, to guarantee real time
keyframe_parallax: 10.0 # keyframe selection threshold (pixel)

        #imu parameters       The more accurate parameters you provide, the better performance
# TODO: figure out these parameters
acc_n: 0.016 # 0.008 # 0.014           # accelerometer measurement noise standard deviation. #0.001372
gyr_n: 0.00024 # 0.00012 # 0.0006          # gyroscope measurement noise standard deviation.     #0.000061087
acc_w: 0.0001 # 0.00004 # 0.000277          # accelerometer bias random work noise standard deviation.  #0.000027719
gyr_w: 0.00001 #0.000004 # 0.00003428        # gyroscope bias random work noise standard deviation.     #0.000003428
g_norm: 9.81007     # gravity magnitude


            #loop closure parameters
loop_closure: 0 # 1   #if you want to use loop closure to minimize the drift, set loop_closure true and give your brief pattern file path and vocabulary file path accordingly;
            #also give the camera calibration file same as feature_tracker node
pattern_file: "/support_files/brief_pattern.yml"
voc_file: "/support_files/brief_k10L6.bin"
min_loop_num: 25
